File: ride-service/src/main/java/com/example/rideservice/config/OpenWeatherMapClientConfig.java
package com.example.rideservice.config;

import com.github.prominence.openweathermap.api.OpenWeatherMapClient;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class OpenWeatherMapClientConfig {


    @Value("${openweather.key}")
    private String KEY;

    @Bean
    public OpenWeatherMapClient openWeatherMapClient() {
        return new OpenWeatherMapClient(KEY);
    }
}

File: price-calculator-service/src/main/kotlin/org/example/pricecalculatorservice/config/OpenWeatherMapClientConfig.kt
package com.example.rideservice.config

import com.github.prominence.openweathermap.api.OpenWeatherMapClient
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class OpenWeatherMapClientConfig {

    @Value("\${openweather.key}")
    private lateinit var key: String

    @Bean
    fun openWeatherMapClient(): OpenWeatherMapClient {
        return OpenWeatherMapClient(key)
    }
}
